_posts/2020-03-16-dask-usage-at-blue-yonder.markdown

```diff
@@ -0,0 +1,161 @@
+endpoints independent of the actual container instance.
+
+Running on top of a system like Mesos or Kubernetes provides us with resilience since a failing worker 
+(for instance, as result of a failing hardware node)
+can simply be restarted on another node of the system. 
+It also enables us to easily commission or decommission Dask clusters, making the amount of clusters we run
+manageable in the first place.
+
+Running 500 Dask clusters also requires a lot of hardware. We have put two measures in place to improve
+the utilization of hardware resources: oversubscription and autoscaling.
+
+### Oversubscription
+
+[Oversubscription](http://mesos.apache.org/documentation/latest/oversubscription/) is a feature of Mesos
+that allows allocating more resources than physically present to services running on the system.
+This is based on the assumption that not all services exhaust all of their allocated resources at the same time.
+If the assumption is violated, we prioritize the resouces to the more important ones.
+We use this to re-purpose the resources allocated for production clusters but not utilized the whole time
+ and use them for development and staging systems.
+
+### Autoscaling
+
+Autoscaling is a mechanism we implemented to dynamically adapt the number of workers in a Dask cluster
+to the load on the cluster. This is possible since Mesos/Kubernetes . This
+makes it really easy to add or remove worker instances from an existing Dask cluster.
+
+To determine the optimum number of worker instances to run, we added the ``desired_workers`` metric to Distributed.
+The metric exposes
+the degree of parallelism that a computation has and thus allows us to infer how much workers a cluster should
+ideally have. Based on this metric, as well as on the overall resources available and on fairness criteria
+(remember, we run a lot of Dask/Distributed clusters), we add or remove workers to our clusters.
+To resolve the problem of balancing the conflicting requirements for different resouces like RAM or CPUs
+ using [Dominant Resource Fairness](https://cs.stanford.edu/~matei/papers/2011/nsdi_drf.pdf).
+
+## Dask issues
+
+The particular way we use Dask, especially running it in containers connected by reverse proxies and the fact
+that we dynamically add/remove workers from a cluster quite frequently for autoscaling has lead us to hit some
+edge cases and instabilities and given us the chance to contribute some fixes and improvements to Dask.
+For instance, we were able to 
+[improve stability after connection failures](https://github.com/dask/distributed/pull/3246) or when 
+[workers are removed from the cluster](https://github.com/dask/distributed/pull/3366). 
+
+If you are interested in our contributions to Dask and our commitment to the dask community, please
+also check out our blog post [Karlsruhe to D.C. ― a Dask story](2020-03-24-dask-developer-workshop.markdown).
```

Comment:
```suggestion
also check out our blog post [Karlsruhe to D.C. ― a Dask story](dask-developer-workshop/).
```
I think it's just this? We just changed the date before publishing, so otherwise it'd be `2020-05-28`. You may want to rebase and check using the jekyll commands